src/libpostalParser.js

```diff
@@ -0,0 +1,44 @@
+var logger = require('pelias-logger').get('text-analyzer');
+
+// mapping object from libpostal fields to pelias fields
+var field_mapping = {
+  category:       'category',
+  house:          'query',
+  house_number:   'number',
+  road:           'street',
+  suburb:         'neighbourhood',
+  city_district:  'borough',
+  city:           'city',
+  state_district: 'county',
+  state:          'state',
+  postcode:       'postalcode',
+  country:        'country'
+};
+
+// wrapper for libpostal that injects the actual parse function for easier
+// testing purposes
+module.exports.create = function create(parse_address) {
+  return {
+    parse: function parse(query) {
+      // call the parsing function (libpostal)
+      var parsed = parse_address(query);
```

Comment:
should check that parse_address is indeed a function before calling it. maybe best to do it at the top of the create function and throw an exception if it's not a function or expects the wrong number of parameters.